src/test_functions/test_functions.rs

```diff
@@ -0,0 +1,507 @@
+use std::f64;
+
+
+/// # Rastrigin function
+/// n-dimensional \
+/// Global minimum: \
+/// f(0, ..., 0) = 0
+pub fn rastrigin(x: &Vec<f64>) -> f64 {
+    if x.len() < 1 {
+        panic!("Rastrigin function takes an at least one dimensional vector as a parameter.");
+    }
```

Comment:
It would be more clear to just use `assert!` macro here. 
This can be applied to parameter validation you did in every of the implemented functions.